### PR TITLE
Adding Jekyll pages api search

### PR DESCRIPTION
**Pull request by line47**
Integrating the search with existing functionality. 
https://github.com/18F/jekyll_pages_api_search/

Thanks for the help @afeld 🎉

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,2 +1,6 @@
 source "https://rubygems.org"
 gemspec :name => 'uswds-jekyll'
+
+group :jekyll_plugins do
+  gem 'jekyll_pages_api_search'
```
Comment (line47) [inline Gemfile]: Adding the gem

File: _config.yml [hunk 2]
```diff
@@ -1 +1,19 @@
 title: This is the site title
+
```
Comment (line47) [inline _config.yml]: Adding config for search

File: _data/header.yml [hunk 3]
```diff
@@ -14,9 +14,8 @@ secondary:
...
-  # 'param' keys, e.g.
+  # 'query' keys, e.g.
   #
-  # search:
-  #   action: /search/
-  #   param: q
-  search: false
+  search:
+    action: /search/
+    query: q
```
Comment (line47) [inline _data/header.yml]: changing `param` to `query` as query is what is used in the form code.

File: _includes/scripts.html [hunk 4]
```diff
@@ -15,3 +15,12 @@
   {% assign _src = _script.src | default: _script %}
 <script src="{{ _src | replace: _async_marker, '' | relative_url }}"{% if _script.async or _src contains _async_marker %} async{% endif %}></script>
 {% endfor %}{% endfor %}
+
+{% if page.layout == 'search-results' %}
```
Comment (line47) [inline _includes/scripts.html]: Including scripts when the `search-results` layout is used

File: _layouts/search-results.html [hunk 5]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (line47) [inline _layouts/search-results.html]: Adding search results layout

File: _includes/scripts.html [hunk 6]
```diff
@@ -15,3 +15,12 @@
...
+<script>
+var JEKYLL_PAGES_API_SEARCH_UI_OPTIONS = {
+  inputElementId: 'search-field-small'
```
Comment (afeld) [inline _includes/scripts.html]: The default uses ID `search-input` - if you change your markup to use that ID instead, you can exclude this `<script>` section.

File: Gemfile (hunk 1 above)
Comment (afeld) [reply Gemfile]: You will need to tell users to do this in the README. Adding it to the `gemspec`(s?) is better, but that won't automatically add it to the `jekyll_plugins` `group`.

File: _includes/scripts.html (hunk 6 above)
Comment (line47) [reply _includes/scripts.html]: Thanks, I've made that update 👍

**Comment by line47** [comment]
Hey @afeld I'm testing this on the https://github.com/18F/frontend guide and I'm close to getting it working I have the search results page loading but I'm not seeing any results.

**Comment by line47** [comment]
@afeld if you can give a second pass at this review that would be great!
I've addressed all the issues I believe and you can see the [frontend-guide](https://github.com/18F/frontend/pull/181) updates as a test.

**Comment by hursey013** [comment]
@line47 I added search in the same way that you've added it here to [another project](https://github.com/GSA/plainlanguage.gov/pull/48/files) and it appears to be working great. The only thing that may improve it is to remove the `*.md` files from a the `assets/uswds/` folder since they are being included in the search results and my attempts to ignore those files via jekyll don't seem to have any effect.

**Comment by hursey013** [comment]
Actually scratch that, it looks like you can ignore that uswds directory using defaults in `_config.yml` https://github.com/GSA/plainlanguage.gov/commit/f69d5ddaf779e8ee2807509b995e150c28116c17

**Comment by line47** [comment]
Thanks @hursey013 I just updated to skip indexing on the `assets/uswds` dir. 🎉 
Was there documentation on the https://github.com/18F/jekyll_pages_api_search repo? If so, I don't see it anywhere.

**Comment by hursey013** [comment]
Yeah I think the only documentation on excluding content with`jekyll_pages_api_search` right now is the bit about being able to use `skip_index: true` on individual pages. Using [defaults](https://jekyllrb.com/docs/configuration/#front-matter-defaults) within the config just allow us to apply that value to entire directories which is more convenient that adding it to each page. In this case, it's not even possible to add it to the markdown files in `assets/uswds` since those are included by the theme gem. I guess you could probably use this approach if you have a site with a lot of content you didn't want indexed and apply `skip_index: true` to everything in the path, and then just whitelist pages you do want included with `skip_index: false`since I _think_ the value in the the actual file will take precedent over the default value.

File: README.md [hunk 7]
```diff
@@ -1,31 +1,37 @@
+---
+skip_index: true
```
Comment (hursey013) [inline README.md]: I _think_ this may be unnecessary since I don't think this file is ever included into the site that is using the theme gem.

**Pull request merged**